tapdb/universe_forest.go

```diff
@@ -125,3 +128,36 @@ func (b *BaseUniverseForest) RootNodes(
 
 	return uniRoots, nil
 }
+
+// RegisterIssuance inserts a new minting leaf within the universe
+// tree, stored at the base key.
+func (b *BaseUniverseForest) RegisterIssuance(ctx context.Context,
+	id universe.Identifier, key universe.BaseKey,
+	leaf *universe.MintingLeaf,
+	metaReveal *proof.MetaReveal) (*universe.IssuanceProof, error) {
+
+	var (
+		writeTx BaseUniverseStoreOptions
+
+		err           error
+		issuanceProof *universe.IssuanceProof
+	)
+
+	dbErr := b.db.ExecTx(
+		ctx, &writeTx, func(dbTx BaseUniverseForestStore) error {
+			issuanceProof, err = universeRegisterIssuance(
+				ctx, dbTx, id, key, leaf, metaReveal,
+			)
+
+			// TODO(ffranr): update universe forest
```

Comment:
Might be useful to have it return stuff like the new universe root for that asset (the above func) so this can then be inserted into the multiverse tree.

Reply:
It was included in the returned issuance proof, but I think it's a good idea to return it separately also. :+1: